test/import/FSHImporter.context.test.ts

```diff
@@ -678,14 +685,38 @@ describe('FSHImporter', () => {
         'anteater'
       ]);
       expect(codeSystem.rules[2].sourceInfo.file).toBe('Zoo.fsh');
-      assertCodeCaretRule(
+      assertCaretValueRule(
         codeSystem.rules[3],
-        ['anteater', 'northern'],
+        '',
         'property[0].valueString',
-        'They are strong climbers.'
+        'They are strong climbers.',
+        false,
+        ['anteater', 'northern']
       );
       expect(codeSystem.rules[3].sourceInfo.file).toBe('Zoo.fsh');
       expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
     });
+
+    it('should parse a rule set that uses an indented CaretValueRule on a top-level concept', () => {
+      const input = `
+      RuleSet: ZOO
+      * #anteater "Anteater"
+        * ^property[0].valueString = "Their threat pose is really cute."
+      `;
+      const result = importSingleText(input, 'Zoo.fsh');
+      const codeSystem = result.ruleSets.get('ZOO');
+      assertConceptRule(codeSystem.rules[0], 'anteater', 'Anteater', undefined, []);
+      expect(codeSystem.rules[0].sourceInfo.file).toBe('Zoo.fsh');
+      assertCaretValueRule(
+        codeSystem.rules[1],
+        'anteater',
+        'property[0].valueString',
+        'Their threat pose is really cute.',
+        false,
+        ['anteater']
+      );
```

Comment:
There is a bit of a problem with the rule here: its `isCodeCaretRule` property is false, since the rule gets parsed by `caretValueRule`. I think that when parsing a RuleSet, it may be necessary to check what the previous rule was in order to correctly determine what kind of context applies.

Reply:
Yeah this brings up a good point. So at this point in the code, the `isCodeCaretRule` flag isn't actually necessary. That property is only used in the `toFSH` method of `CaretValueRule`, so specifically it is relevant to creating preprocessed FSH. So that flag is only set in the `CodeSystemExporter`, so that by the time the `toFSH` method is called, the flag is set. Because you're right, on the import side, it is hard to tell which context applied without checking adjacent rules.

I _think_ the approach should still work fine, but I feel like the name of the flag is pretty confusing, since for much of the lifespan of the `CaretValueRule`, the flag is always `false`. Maybe I could change it to something like `printAsCodeCaretRule`? Any other ideas?

Reply:
Ah, okay, I think you're right that it still ends up fine in terms of the FHIR output. Since a `CodeSystem` can't have a `CaretValueRule` with a regular path (since it doesn't have any elements to reference with a regular path), it necessarily wants to use the path parts as codes. Or put another way, every `CaretValueRule` that ends up on a `CodeSystem` is a code caret rule. And since the writing of preprocessed FSH only happens after export, and that export will set the flag, it'll end up fine in that part of the output as well.

This may end up making some exciting things happen once insert rules can be indented, though, so I'll have to look carefully at that PR.